Mage/Module.cs

```diff
@@ -8,29 +8,39 @@
 
 namespace Wizcorp.MageSDK.MageClient
 {
-
-	public class Module<T> : Singleton<T> where T : class, new()
+	using MageCommandAction = Action<JObject, Action<Exception, JToken>>;
+	using MageCommandFunction = Func<JObject, UserCommandStatus>;
+	
+	public abstract class Module<T> : Singleton<T> where T : class, new()	
 	{
-		//
+		// Tells us if the module was set up
+		protected bool SetupCompleted = false;
+
+		// Mage singleton accessor
 		protected Mage Mage
 		{
 			get { return Mage.Instance; }
 		}
 
+		// Contextualized logger
 		protected Logger Logger
 		{
 			get { return Mage.Logger(GetType().Name); }
 		}
 
-
-		//
+		// Static
 		protected virtual List<string> StaticTopics
 		{
-			get { return null; }
+			get { return new List<string> {}; }
```

Comment:
I assume a null list would still allow you to execute list API calls, whereas null would crash if you do not check for null. Since neither the IDE nor the compiler warns against nullable types, I thought this would be safer to do this.